documentation/modules/oauth/con-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,452 @@
+
+```
+bin/kafka-console-producer.sh --broker-list my-cluster-kafka-bootstrap:9092 --topic my-topic \
+  --producer.config=/tmp/team-a-client.properties
+First message
+```
+
+When we press `Enter` to push the first message we receive `Not authorized to access topics: [my-topic]` error.
+
+`team-a-client` has a `Dev Team A` role which gives it permissions to do anything on topics that start with 'a_', and only write to topics that start with 'x_'.
+The topic named `my-topic` matches neither of those.
+
+Use CTRL-C to exit the CLI application, and let's try to write to topic `a_messages`.
+
+```
+bin/kafka-console-producer.sh --broker-list my-cluster-kafka-bootstrap:9092 --topic a_messages \
+  --producer.config /tmp/team-a-client.properties
+First message
+Second message
+```
+
+Although we can see some unrelated warnings, looking at the Kafka container log there is DEBUG level output saying 'Authorization GRANTED'.
+
+Use CTRL-C to exit the CLI application.
+
+You can see the Kafka container log by running:
+
+    kubectl logs my-cluster-kafka-0 -f
+
+### Consuming the Messages
+
+Let's now try to consume the messages we have produced.
+
+    bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic a_messages \
+      --from-beginning --consumer.config /tmp/team-a-client.properties
+
+This gives us an error like: `Not authorized to access group: console-consumer-55841`.
+
+The reason is that we have to override the default consumer group name - `Dev Team A` only has access to consumer groups that have names starting with 'a_'.
+Let's set custom consumer group name that starts with 'a_'
+
+    bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic a_messages \
+      --from-beginning --consumer.config /tmp/team-a-client.properties --group a_consumer_group_1
+
+We should now receive all the messages for the 'a_messages' topic, after which the client blocks waiting for more messages.
+
+Use CTRL-C to exit.
+
+
+### Using Kafka's CLI Administration Tools
+
+Let's now list the topics:
+
+    bin/kafka-topics.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --command-config /tmp/team-a-client.properties --list
+
+We get one topic listed: `a_messages`.
+
+Let's try and list the consumer groups:
+
+    bin/kafka-consumer-groups.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --command-config /tmp/team-a-client.properties --list
+
+Similarly to listing topics, we get one consumer group listed: `a_consumer_group_1`.
+
+There are more CLI administrative tools. For example we can try to get the default cluster configuration:
+
+    bin/kafka-configs.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --command-config /tmp/team-a-client.properties \
+      --entity-type brokers --describe --entity-default
+
+But that will fail with `Cluster authorization failed.` error, because this operation requires cluster level permissions which `team-a-client` does not have.
```

Comment:
```suggestion
Fetching the default cluster configuration fails cluster authorization, because the operation requires cluster level permissions that `team-a-client` does not have.

[source,shell]
----
bin/kafka-configs.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --command-config /tmp/team-a-client.properties \
  --entity-type brokers --describe --entity-default
----

But that will fail with `Cluster authorization failed.` error, because this operation requires cluster level permissions which `team-a-client` does not have.
```